testcases/PowerNVDump.py

```diff
@@ -149,7 +150,40 @@ def setup_test(self, dump_place="local"):
                 "ls -l /var/crash | grep '^d'| awk '{print $9}'")
         if dump_place == "net":
             self.crash_content = self.c.run_command(
-                "ssh root@%s \"ls -l %s | grep '^d'\" | awk '{print $9}'" % (self.dump_server_ip, self.dump_path))
+                "ssh -i %s root@%s \"ls -l %s | grep '^d'\" | awk '{print $9}'" % (self.rsa_path, self.dump_server_ip, self.dump_path))
+            log.debug("crash content is %s" % (self.crash_content))
+
+    def setup_pwdless_auth(self):
+        # Setup Pulickey authentication for Kdump over ssh and NFS
+        # creates key using ssh-keygen and copies key using sshpass
+        # check for the successful ssh connection
+        self.c.run_command("rm -f %s" % self.rsa_path)
+        if self.distro == "rhel":
+            try:
+                self.c.run_command("yum -y install sshpass")
+            except CommandFailed:
+                self.fail("sshpass package not available in Repository")
+        elif self.distro == "sles":
+            try:
+                self.c.run_command("zypper install -y sshpass")
+            except CommandFailed:
+                self.fail("sshpass package not available in Repository")
+        else:
+            try:
+                self.c.run_command("apt-get install -y sshpass")
+            except CommandFailed:
+                self.fail("sshpass package not available in Repository")
```

Comment:
multiple similar fail logging.. can this be optimised ?

Reply:
yeah but not sure as we have different distros

Reply:
@PraveenPenguin do you have any hints ?

Reply:
Other way can be:
              if self.distro == "rhel":
                      cmd = "yum -y install sshpass"
              elif self.distro == "sles":
                     cmd = "zypper install -y sshpass"
              else:
                     cmd = "apt-get install -y sshpass"
  try:
       self.c.run_command(cmd)
  except CommandFailed: 
             self.fail("sshpass package not available in Repository")

Reply:
thank you @sacsant, will incorporate this